src/sync.rs

```diff
@@ -27,20 +29,46 @@ pub fn sync(config: &crate::Config) -> crate::Result<()> {
                 crate::Backend::FTD => {
                     crate::ftd::handle(file, config.root.as_str(), config.collection.as_str())?
                 }
-                crate::Backend::Raw => {
-                    crate::raw::handle(file, config.root.as_str(), config.collection.as_str())?
-                }
+                crate::Backend::Raw => crate::raw::handle(
+                    &tree,
+                    file,
+                    config.root.as_str(),
+                    config.collection.as_str(),
+                )?,
             });
         }
 
+        if config.backend.is_raw() {
+            let readme_content = if let Some(readme) = tree.readme() {
+                let file = crate::FileMode::Modified(readme);
+                Some(file.content()?)
+            } else {
+                None
+            };
+
+            let collection_toc =
+                tree.collection_toc(config.root.as_str(), config.collection.as_str());
+
+            actions.push(ft_api::bulk_update::Action::Updated {
+                id: config.collection.to_string(),
+                content: format!(
+                    "-- h1: {}\n\n\n{}\n{}\n\n\n{}",
```

Comment:
Dont do this. Use `ftd` to create this. You will have to enable "fifthtry" feature now.

Reply:
@amitu what should we use for `index-extra`, We have to prepend it and it is in string.

Reply:
Parse index-extra using `ftd` and concatenate the two vectors (index_extra_vec + optional_title + README.md_ftd + toc).